File: Contests/CodeforcesPython/Codeforces Round 681 Div. 2 based on VK Cup 20192020  Final/C_The_Delivery_Dilemma/C_The_Delivery_Dilemma.py
import sys

# region fastio
input = lambda: sys.stdin.readline().rstrip()
sint = lambda: int(input())
mint = lambda: map(int, input().split())
ints = lambda: list(map(int, input().split()))
# endregion fastio

# MOD = 998_244_353
# MOD = 10 ** 9 + 7
# DIR4 = ((-1, 0), (0, 1), (1, 0), (0, -1)) #URDL
# DIR8 = ((-1, 0), (-1, 1), (0, 1), (1, 1), (1, 0), (1, -1), (0, -1), (-1, -1))

def solve() -> None:
    n = sint()
    A = ints()
    B = ints()
    ans = s = sum(B)
    idx = sorted(range(n), key = lambda x: A[x])
    for i in idx:
        s -= B[i]
        ans = min(ans, max(A[i], s))
    print(ans)

for _ in range(int(input())):
    solve()

File: Contests/CodeforcesPython/Codeforces Round 681 Div. 2 based on VK Cup 20192020  Final/C_The_Delivery_Dilemma/C_The_Delivery_Dilemma.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
	"sort"
)

func solve(_r io.Reader, _w io.Writer) {
	in := bufio.NewReader(_r)
	out := bufio.NewWriter(_w)
	defer out.Flush()

	var T, n int
	for Fscan(in, &T); T > 0; T-- {
		Fscan(in, &n)
		var ans, s int
		idx := make([]int, n)
		a := make([]int, n)
		for i := range a {
			Fscan(in, &a[i])
			idx[i] = i
		}
		b := make([]int, n)
		for i := range b {
			Fscan(in, &b[i])
			ans += b[i]
			s += b[i]
		}

		sort.Slice(idx, func(i, j int) bool {
			return a[idx[i]] <= a[idx[j]]
		})

		for _, i := range idx {
			s -= b[i]
			ans = min(ans, max(a[i], s))
		}

		Fprintln(out, ans)
	}
}

func main() { solve(os.Stdin, os.Stdout) }
func min(a, b int) int {
	if a <= b {
		return a
	} else {
		return b
	}
}
func max(a, b int) int {
	if a >= b {
		return a
	} else {
		return b
	}
}
